lib/SILOptimizer/Mandatory/MandatoryCombine.cpp

```diff
@@ -107,14 +107,14 @@ class MandatoryCombiner final
     if (!areAllValuesTrivial(fullApplyArguments, *function)) {
       return nullptr;
     }
-    auto partialApplyArguments = ApplySite(partialApply).getArguments();
+
+    ApplySite partialApplySite(partialApply);
+    auto partialApplyArguments = partialApplySite.getArguments();
     if (!areAllValuesTrivial(partialApplyArguments, *function)) {
       return nullptr;
     }
 
-    auto callee = partialApply->getCallee();
-
-    ApplySite partialApplySite(partialApply);
+    SILValue callee = partialApply->getCallee();
```

Comment:
What guides your style as far as using auto vs not in cases like these?

Reply:
I prefer to use SILValue instead of auto. B/c I really want to use auto for pointers or types with large names. SILValue isn't too long and I think provides semantic meaning that is useful.

Reply:
That is just my style opinion though. People do it differently. Generally I try to only use auto if the type is on the same line.

Reply:
E.x.:

```
if (auto *x = dyn_cast<SILInstruction>(foo)) { ... }
```